### PR TITLE
Fix: Input/Radio should be disabled when disabling the formcontrol

### LINKED ISSUE #1171: [input @alpha.8-1] Setting bound FormControl to disabled causes valueAccessor.setDisabledState error
#### Bug, feature request, or proposal:

Bug
#### What is the expected behavior?

Angular recommends, when using reactive forms, to set `disabled` on the `FormControl` rather than the input element in the template:

#### What is the current behavior?

Setting `disabled: true` on a `FormControl` bound to `md-input` will cause the app to crash with error:

> TypeError: this.valueAccessor.setDisabledState is not a function

This does not happen to the native text input.
#### What are the steps to reproduce?

[Open this plunker ](http://plnkr.co/edit/0Pw4kwS6jJ8CI5nKh5On?p=preview)and look at the console. If in `AppComponent` you replace:

```
nameCtrl = new FormControl({value:'Jane', disabled:true});
```

With

```
nameCtrl = new FormControl('Jane');
```

The error will go away
#### Which versions of Angular, Material, OS, browsers are affected?

`Angular 2 rc 6` with `Material alpha 8.1`

### PR DESCRIPTION
**Pull request by Sabartius**
Fixes #1171 for Input-Elements and Radio

@jelbourn @kara I've tried to repair the checkbox too, but i couldn't test it properly.
My test for the checkbox only worked, when i deleted the ChangeDetectionStrategy.OnPush from the Component, but thats not the solution i wanted to push.

### CODE REVIEW

**Comment by Sabartius** [comment]
I tried the checkbox in an demo app and it disables & enables successfully. So it's only something with the change-detection in the testenvironment that's not working as expected.

For Radio-Buttons i couldn't even get the reactiveForms to work with them. Maybe you could give me a hint how i need to configure the radio-group to set the Value of the FormControl?

File: src/lib/input/input.spec.ts [hunk 1]
```diff
@@ -608,6 +616,39 @@ describe('MdInput', function () {
...
+
+      formControl.disable();
+      fixture.detectChanges();
+
+      fixture.whenStable().then(() => {
+        // Temporary workaround, see https://github.com/angular/angular/issues/10148
+        fixture.detectChanges();
+        fixture.whenStable().then(() => {
+          expect(inputElement.disabled).toBeTruthy('InputElement should be disabled');
+          expect(mdInput.disabled).toBeTruthy('Component should be disabled');
```
Comment (jelbourn) [inline src/lib/input/input.spec.ts]: Huh, I had no idea you could pass a message like that.

File: src/lib/radio/radio.spec.ts [hunk 2]
```diff
@@ -424,6 +432,48 @@ describe('MdRadio', () => {
...
+    let fixture: ComponentFixture<RadioGroupWithReactiveForms>;
+    let formControl: FormControl;
+    let radioInputs: Array<MdRadioButton> = [];
+    let nativeRadioInputs: Array<HTMLInputElement> = [];
+
+    beforeEach(async(() => {
+      fixture = TestBed.createComponent(RadioGroupWithReactiveForms);
+      fixture.detectChanges();
+
+      fixture.whenStable().then(() => {
```
Comment (jelbourn) [inline src/lib/radio/radio.spec.ts]: Why do you need a `whenStable` in the `beforeEach`?

File: src/lib/radio/radio.spec.ts [hunk 3]
```diff
@@ -424,6 +432,48 @@ describe('MdRadio', () => {
...
+        // Temporary workaround, see https://github.com/angular/angular/issues/10148
+        fixture.detectChanges();
+        fixture.whenStable().then(() => {
+          radioInputs.forEach(input =>
+              expect(input.disabled).toBeTruthy(input.value + ' should be disabled'));
+          nativeRadioInputs.forEach(input =>
+              expect(input.disabled).toBeTruthy('Native ' + input.id + ' should be disabled'));
```
Comment (jelbourn) [inline src/lib/radio/radio.spec.ts]: You can use template strings for these messages

**Comment by jelbourn** [comment]
@kara should take a look

File: src/lib/input/input.spec.ts (hunk 1 above)
Comment (Sabartius) [inline src/lib/input/input.spec.ts]: It helps alot when looking for errors :)

File: src/lib/radio/radio.spec.ts (hunk 2 above)
Comment (Sabartius) [inline src/lib/radio/radio.spec.ts]: Didn't need that one, got removed in the next commit

File: src/lib/radio/radio.spec.ts (hunk 3 above)
Comment (Sabartius) [inline src/lib/radio/radio.spec.ts]: Done

**Comment by Sabartius** [comment]
@jelbourn @kara what's your opinion on the checkbox? I can push a solution, even a test for the checkbox, but because of the changedetection-strategy of the checkbox-component i cannot test if the underlying input-element gets disabled, only that the component gets disabled.

**Comment by jelbourn** [comment]
@Sabartius can you push the change for checkbox including the failing test? We can take a look at see what needs to be changed.

**Comment by Sabartius** [comment]
@jelbourn Pushed it. Hope you can find the problem :)

**Comment by jelbourn** [comment]
@tinayuangao can you follow up on this?

**Comment by jelbourn** [comment]
Superseded by #1750

**Pull request closed without merging**